secrets/secrets.go

```diff
@@ -323,19 +324,30 @@ func secretsValidate(secretsDocument Document) (*Secrets, error) {
 
 // walkCSIDirectory parses a directory for vault secrets and merges them into one object
 func walkCSIDirectory(dir fs.FS) (Document, error) {
+	const (
+		// This is where k8s actually writes the content,
+		// ref: https://pkg.go.dev/sigs.k8s.io/secrets-store-csi-driver/pkg/util/fileutil#AtomicWriter.Write
+		k8sSubdirectory = "..data"
+	)
 	secretsDocument := Document{
 		Secrets: make(map[string]GenericSecret),
 	}
 	err := fs.WalkDir(
 		dir,
-		".",
+		k8sSubdirectory,
```

Comment:
🔕 Probably worth a quick stat to make sure it exists, and if it doesn't, return an actionable error ("dir %q does not appear to be the root of a Vault CSI mount point" or something).

Alternately you can probably check the error from WalkDir to see if it's a PathError that's NotExists and matches the k8sSubdirectory.

Reply:
[that's basically the first thing `fs.WalkDir` does](https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/io/fs/walk.go;l=110), and the error returned will be returned to the caller on https://github.com/reddit/baseplate.go/blob/2dd3ab82ca91dabd8b2aead5574f57ec30fb2b69/secrets/secrets.go#L355, so that should be good enough? I don't feel like we should add a special stat call to this :)

Reply:
My concern is the error message, given that we're pointing WalkDir at a directory that the user did not specify directly for an "implementation detail" reason.  `os.Stat("/path/to/your/dir/..data"): file does not exist` is going to be a very opaque error message, and it kinda looks like someone forgot a slash after a `..`.

However, `dir "/path/to/your/dir" does not appear to be the root of a Vault CSI mount point: os.Stat("/path/to/your/dir/..data"): file does not exist` or something is going to be much easier to use to figure out what you need to do -- i.e. point it at a Vault CSI mount point, not any old dir.

Reply:
Done in latest fixup commit.